posthog/api/__init__.py

```diff
@@ -73,12 +73,25 @@ def api_not_found(request):
 
 # Project nested endpoints
 projects_router = router.register(r"projects", team.TeamViewSet, "projects")
-projects_router.register(
-    r"event_definitions", event_definition.EventDefinitionViewSet, "project_event_definitions", ["team_id"],
-)
-projects_router.register(
-    r"property_definitions", property_definition.PropertyDefinitionViewSet, "project_property_definitions", ["team_id"],
-)
+if is_ee_enabled():
```

Comment:
@timgl Since we said that `settings.EE_AVAILABLE` aren't the correct settings to see if the user has an enterprise account, what are the settings I should be looking for/creating ?

Reply:
@Twixes 🥺

Reply:
@liyiy We can just get rid of this and always import these routes, then just override them if the ee folder is available

Reply:
:wave:
Actually you _should_ use `settings.EE_AVAILABLE` when registering routers on this line (as this feature, as I understand, is not part of our free open-source offering – only premium).

As for `is_ee_enabled`, it's been renamed to `is_clickhouse_enabled` in `master` (make sure to rebase this PR), because it only says whether ClickHouse is used instead of Postgres for analytics. If you want to add a premium feature **regardless of analytics DB**, you should add that feature (here, let's say `"taxonomy"`) to `ee.models.License.ENTERPRISE_FEATURES`. That way all existing premium customers will get it.

Then in viewsets and serializers which pertain to this premium feature, you should check `organization.is_feature_available("taxonomy")` and if false raise an error saying that this feature is not available (429?). This may be done somewhere already, @paolodamico likely has more context on handling of plans.

Reply:
> Actually you should use settings.EE_AVAILABLE when registering routers on this line (as this feature, as I understand, is not part of our free open-source offering – only premium).

Yeah but that's not how Li and I wrote it. we already override the routes if the ee folder is available (`ee/api/event_definition.py`),  and then check within that view if the user has the `feature_available`

So @liyiy you should just remove the if statement all together for this specific line

Reply:
@timgl  I don't know how to override a route 😦  I've added the "enterprise" definition routes to ee's `def extend_api_router` but they don't override the existing definition ones in `posthog/api/__init__` 

@Twixes The feature is added to the enterprise plans list, do you mean I should be handling this in the actual viewsets/serializers itself instead of with routes? Like I should be pointing at the EnterpriseEventDefinition viewset if that conditional check is met?

Reply:
> @timgl I don't know how to override a route 😦 I've added the "enterprise" definition routes to ee's def extend_api_router but they don't override the existing definition ones in posthog/api/__init__

Ah in that case we should have an if statement, but not use `is_clickhouse_enabled` (as this has nothing to do with clickhouse). I think your initial approach was right.

Reply:
> > @timgl I don't know how to override a route 😦 I've added the "enterprise" definition routes to ee's def extend_api_router but they don't override the existing definition ones in posthog/api/**init**
> 
> Ah in that case we should have an if statement, but not use `is_clickhouse_enabled` (as this has nothing to do with clickhouse). I think your initial approach was right.

Right, sorry, so in that case, what am I checking with since `ee_enabled` isn't the correct setting? (EE condition is still true for me even when I don't have a license key on my account) How can I check the user's "enterprise plan feature" setting in the route? Let me know if my question doesn't make any sense 😅

Reply:
never mind I think I got it

Reply:
had a quick check in with @EDsCODE and I'm going to try some other things: 1) backfilling the enterprise definitions table with the non-enterprise definitions and 2) sending enterprise users to the enterprise viewset instead of trying to do it in the routes (which doesn't work)